redash/query_runner/presto.py

```diff
@@ -76,9 +78,18 @@ def run_query(self, query):
 
         cursor = connection.cursor()
 
+        # If a column appears more than once with the same name, subsequent
+        # instances are numbered 1, 2, , ... N.
+        columns_seen = defaultdict(int)
+        def make_unique(name):
+          columns_seen[name] += 1
+          if columns_seen[name] == 1:
+            return name
+          return name + str(columns_seen[name] - 1)
+
         try:
             cursor.execute(query)
-            columns_data = [(row[0], row[1]) for row in cursor.description]
+            columns_data = [(make_unique(row[0]), row[1]) for row in cursor.description]
```

Comment:
Actually you can use the [`fetch_columns` method](https://github.com/getredash/redash/blob/master/redash/query_runner/__init__.py#L74) this class inherits from `BaseQueryRunner`. 

In the case of Presto, it will be something like this:
```python
columns = self.fetch_columns([(i[0], PRESTO_TYPES_MAPPING.get(i[1], None)) for i in cursor.description])
```

As a side note, your implementation of deduplication logic is nicer than the one in the `fetch_columns` method, so feel free to replace it there. :+1: